frontend/src/actions/habit_actions.js

```diff
@@ -41,34 +48,64 @@ export const removeHabit = habitId => ({
 
 export const fetchHabits = () => dispatch => (
     getHabits()
-      .then(habits => dispatch(receiveHabits(habits)))
-      .catch(err => console.log(err))
+      .then(
+        habits => dispatch(receiveHabits(habits)), 
+        err => dispatch(receiveErrors(err.response.data))
+      )
+      .catch(err => {
+        dispatch(receiveErrors(err.response.data));
+      })
   );
 
 export const fetchHabit = id => dispatch => (
     getHabit(id)
-      .then(habits => dispatch(receiveHabit(habits)))
-      .catch(err => console.log(err))
+      .then(
+        habits => dispatch(receiveHabit(habits)),
+        err => dispatch(receiveErrors(err.response.data))
+      )
+      .catch(err => {
+        dispatch(receiveErrors(err.response.data));
+      })
```

Comment:
would you be able to rename the `receiveErrors` functions to `receiveHabitErrors` and `receiveResourceErrors` respectively? You export both of them under the same name.
 
You don't need both the error callback and a catch block, so you could probably just use a catch call (but it's up to your preference which you do). to cite [the MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise):

>Processing continues to the next link of the chain even when a .then() lacks a callback function that returns a Promise object. Therefore, **a chain can safely omit every rejection callback function until the final .catch()**.
>
>Handling a rejected promise in each .then() has consequences further down the promise chain. Sometimes there is no choice, because an error must be handled immediately. In such cases we must throw an error of some type to maintain error state down the chain. On the other hand, in the absence of an immediate need, it is simpler to leave out error handling until a final .catch() statement. A .catch() is really just a .then() without a slot for a callback function for the case when the promise is resolved.

Reply:
yes will do